File: 7-kyu/histogram---h1/python/solution.py
def histogram(results):
    ret = ''
    curr_side = len(results)
    for result in results[::-1]:
        ret += str(curr_side) + "|"
        if result == 0:
            ret += "\n"
            curr_side -= 1
            continue
        ret += ''.join('#' for _ in range(result)) + ' ' + str(result) + '\n'
        curr_side -= 1
    return ret

File: 7-kyu/histogram---h1/c++/solution.cpp
std::string histogram(std::vector<int> results)
{
    std::string ret = "";
    int curr_side = 6;
    for (auto r_it = results.rbegin(); r_it != results.rend(); r_it++) {
        std::string line = std::to_string(curr_side) + "|";
        if (*r_it == 0) {
            line += "\n";
            ret += line;
            curr_side--;
            continue;
        }
        line += std::string(*r_it, '#') + " " + std::to_string(*r_it) + "\n";
        ret += line;
        curr_side--;
    }
    return ret;
}